app/src/main/cpp/WidgetPlacement.cpp

```diff
@@ -0,0 +1,46 @@
+/* -*- Mode: C++; tab-width: 20; indent-tabs-mode: nil; c-basic-offset: 2 -*-
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "WidgetPlacement.h"
+
+namespace crow {
+
+WidgetPlacementPtr
+WidgetPlacement::FromJava(JNIEnv* aEnv, jobject& aObject) {
+  if (!aObject) {
+    return nullptr;
+  }
+
+  jclass clazz = aEnv->GetObjectClass(aObject);
```

Comment:
Maybe think how to cache this since I'm pretty sure this and GetFieldID are expensive? Probably can be done as follow up.